packages/@aws-containers/aws-ecs-builder/lib/extensions/http-load-balancer.ts

```diff
@@ -0,0 +1,60 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as alb from '@aws-cdk/aws-elasticloadbalancingv2';
+import * as cdk from '@aws-cdk/core';
+import { Service } from '../service';
+import { ServiceExtension, ServiceBuild } from './extension-interfaces';
+
+/**
+ * This extension add a public facing load balancer for sending traffic
+ * to one or more replicas of the application container
+ */
+export class HttpLoadBalancerExtension extends ServiceExtension {
+  private loadBalancer!: alb.IApplicationLoadBalancer;
+  private listener!: alb.IApplicationListener;
+
+  constructor() {
+    super('load-balancer');
+  }
+
+  // Before the service is created go ahead and create the load balancer itself.
+  public prehook(service: Service, scope: cdk.Construct) {
+    this.parentService = service;
+
+    this.loadBalancer = new alb.ApplicationLoadBalancer(scope, `${this.parentService.id}-load-balancer`, {
+      vpc: this.parentService.vpc,
+      internetFacing: true,
+    });
+
+    this.listener = this.loadBalancer.addListener(`${this.parentService.id}-listener`, {
+      port: 80,
+      open: true,
```

Comment:
Should this be configurable? If I remember correctly, this caused some controversies in the `ApplicationLoadBalancedFargateService`.

Reply:
IAM policy creation adjusted as suggested